xbmc/addons/include/kodi_inputstream_types.h

```diff
@@ -28,14 +28,77 @@
 
 extern "C" {
 
+  struct INPUTSTREAM
+  {
+    static const unsigned int MAX_INFO_COUNT  = 8;
+
+    char *m_strURL;
+
+    unsigned int m_nCountInfoValues;
+    struct LISTITEMPROPERTY
+    {
+        const char *m_strKey;
+        const char *m_strValue;
+    } m_ListItemProperties[MAX_INFO_COUNT];
+  };
+
+
   struct INPUTSTREAM_INFO
   {
-    int dummy;
+    static const int INPUTSTREAM_INFO_MAX_STREAMS = 8;
+
+    unsigned int m_nCountStreams;
+
+    struct INPUTSTREAM_STREAM
+    {
+      enum eStreamContent
+      {
+        scVIDEO,
+        scAUDIO,
+        scSUBTITLE,
+        scTELETEXT,
+        scPROGRAMM,
+        scRDS
+      } m_streamContent;	                  /*!< @brief (required) codec id of this stream */
+
+      enum eStreamType
+      {
+        stNone,
+        stAC3,
+        stMPEG2AUDIO,
+        stEAC3,
+        stAACADTS,
+        stAACLATM,
+        stDTS,
+        stMPEG2VIDEO,
+        stH264,
+        stDVBSUB,
+        stTEXTSUB,
+        stTELETEXT,
+      } m_streamType;			              /*!< @brief (required) type this stream */
+
+      const int             m_pID;            /*!< @brief (required) physical index */
+      char                  m_language[4];    /*!< @brief ISO 639 3-letter language code (empty string if undefined) */
+
+      unsigned int          m_FpsScale;       /*!< @brief Scale of 1000 and a rate of 29970 will result in 29.97 fps */
+      unsigned int          m_FpsRate;
+      unsigned int          m_Height;         /*!< @brief height of the stream reported by the demuxer */
+      unsigned int          m_Width;          /*!< @brief width of the stream reported by the demuxer */
+      float                 m_Aspect;         /*!< @brief display aspect of stream */
+
+      unsigned int          m_Channels;       /*!< @brief (required) amount of channels */
+      unsigned int          m_SampleRate;     /*!< @brief (required) sample rate */
+      unsigned int          m_BitRate;        /*!< @brief (required) bit rate */
+      unsigned int          m_BitsPerSample;  /*!< @brief (required) bits per sample */
+      unsigned int          m_BlockAlign;
+    } m_Streams[INPUTSTREAM_INFO_MAX_STREAMS];
```

Comment:
I would like to split this. Kodi first requests an array with stream IDs. Then it can query the properties of each stream separately.